### PR TITLE
Upmerge 20240325 common conf dir

### PR DESCRIPTION
**Pull request by tejlmand**
waiting for feedback on https://github.com/gmarull/sdk-find-my/pull/1

### CODE REVIEW

File: cmake/multi_image.cmake [hunk 1]
```diff
@@ -321,54 +321,65 @@ function(add_child_image_from_source)
...
+      set(ACI_NAME_CONF_DIR ${config_dir}/child_image/${ACI_NAME})
+      if (NOT ${ACI_NAME}_CONF_FILE)
+        ncs_file(CONF_FILES ${ACI_NAME_CONF_DIR}
+          BOARD ${ACI_BOARD}
+          # Child image always uses the same revision as parent board.
+          BOARD_REVISION ${BOARD_REVISION}
+          KCONF ${ACI_NAME}_CONF_FILE
+          DTS ${ACI_NAME}_DTC_OVERLAY_FILE
+          BUILD ${CONF_FILE_BUILD_TYPE}
+          SUFFIX ${${ACI_NAME}_FILE_SUFFIX}
```
Comment (nordicjm) [inline cmake/multi_image.cmake:337]: These two options are exclusive, if there is a suffix then it needs to override the build type, it should not append build type and suffix which seems to be the case:
```
cmake -GNinja -DBOARD=nrf52840dk/nrf52840 -DCONF_FILE=prj_base.conf -Dmcuboot_FILE_SUFFIX=base -DCONFIG_BOOTLOADER_MCUBOOT=y ..

-- Found BOARD.dts: /tmp/bb/zephyr/boards/nordic/nrf52840dk/nrf52840dk_nrf52840.dts
-- Found devicetree overlay: /tmp/bb/zephyr/samples/hello_world/child_image/mcuboot/boards/nrf52840dk_nrf52840_base_base.overlay
-- Found devicetree overlay: /tmp/bb/nrf/modules/mcuboot/usb.overlay
-- Generated zephyr.dts: /tmp/bb/zephyr/samples/hello_world/_AA/mcuboot/zephyr/zephyr.dts
-- Generated devicetree_generated.h: /tmp/bb/zephyr/samples/hello_world/_AA/mcuboot/zephyr/include/generated/devicetree_generated.h
-- Including generated dts.cmake file: /tmp/bb/zephyr/samples/hello_world/_AA/mcuboot/zephyr/dts.cmake

warning: FLASH_MAP (defined at subsys/storage/flash_map/Kconfig:10) has direct dependencies FLASH_HAS_DRIVER_ENABLED with value n, but is currently being y-selected by the following symbols:
 - MCUBOOT_DEVICE_SETTINGS (defined at /tmp/bb/bootloader/mcuboot/boot/zephyr/Kconfig:707), with value y, direct dependencies y (value: y)

error: Aborting due to Kconfig warnings

Parsing /tmp/bb/bootloader/mcuboot/boot/zephyr/Kconfig
Loaded configuration '/tmp/bb/zephyr/boards/nordic/nrf52840dk/nrf52840dk_nrf52840_defconfig'
Merged configuration '/tmp/bb/zephyr/samples/hello_world/child_image/mcuboot/prj_base_base.conf'
Merged configuration '/tmp/bb/zephyr/samples/hello_world/child_image/mcuboot/boards/nrf52840dk_nrf52840_base_base.conf'
Merged configuration '/tmp/bb/nrf/subsys/partition_manager/partition_manager_enabled.conf'
Merged configuration '/tmp/bb/zephyr/samples/hello_world/_AA/mcuboot/zephyr/misc/generated/extra_kconfig_options.conf'
```

**Comment by gmarull** [comment]
integrated

File: cmake/multi_image.cmake (hunk 1 above)
Comment (tejlmand) [reply cmake/multi_image.cmake:337]: this is an upstream issue and should be fixed there.

Comment (nordicjm) [reply cmake/multi_image.cmake:337]: Cannot replicate in upstream, feature works fine

**Pull request closed without merging**